src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_code.go

```diff
@@ -91,13 +93,14 @@ func (a *StoragePowerActorCode_I) WithdrawBalance(rt Runtime, amount actor.Token
 
 	h, st := a.State(rt)
 
-	currEntry, found := st.PowerTable()[minerID]
-	if !found {
-		rt.Abort("minerID not found.")
+	ret := st._ensurePledgeCollateralSatisfied(rt)
+	if !ret {
+		rt.Abort("insufficient pledge collateral.")
```

Comment:
This is unnecessary. Locked-up pledge collateral should not count as `AvailableBalance` so the check on 102 already covers this.

Reply:
Given the premise that a PoSt is only valid when miner has sufficient balance in StoragePowerActor, LockedBalance is the pledge collateral being locked and it is updated by _ensurePledgeCollateralSatisfied. LockedBalance may get out of sync with what is required at any point in time and hence the invocation of _ensurePledgeCollateralSatisfied here. I think what is missing is to unlock funds when LockedBalance is more than pledge required in _ensurePledgeCollateralSatisfied.

Reply:
agree. ensure is good safety, it should indeed also unlock as needed. Maybe a more descriptive method name would be nice. 

May also let miner manually unlock if cost is high (this might get called too often).